src/driver/oracle/OracleConnectionOptions.ts

```diff
@@ -23,6 +31,14 @@ export interface OracleConnectionOptions
      */
     readonly driver?: any
 
+    /**
+     * Utilize the thick driver. Starting from oracledb version 6, it's necessary to set this to true when opting for the thick client usage.
+     * Alternatively, an 'OracleThickModeOptions' object can be configured, which is used for the thick mode configuration by passing it to the 'node-oracledb' driver.
+     * For additional information, refer to the details provided in the following link:
+     * (https://node-oracledb.readthedocs.io/en/latest/api_manual/oracledb.html#oracledb.initOracleClient)
+     */
+    readonly thickMode?: boolean | OracleThickModeOptions
```

Comment:
Is this correct?  It is correctly shown as readonly but documented  in the PR as settable. 

In node-oracledb the only way to use Thick mode is to call initOracleClient(), with or without options.

Reply:
@cjbj: I believe the readonly modifier is correct. Because the connection options are just an interface, and they have to be created initially and passed to the datasource. Also, all other fields of this interface are readonly, which can also be set initially.
The thick mode is initialised in the class OracleDriver.